pkg/resiliency/resiliency.go

```diff
@@ -1,132 +1,508 @@
+				Timeout:        t.Timeout,
+				Retry:          t.Retry,
+				CircuitBreaker: t.CircuitBreaker,
+			},
+			CircuitBreakerScope: scope,
+		}
+		r.actorCBCaches[name], err = lru.New(t.CircuitBreakerCacheSize)
+		if err != nil {
+			return err
+		}
+	}
+
+	for name, t := range buildingBlocks.Components {
+		r.components[name] = PolicyNames{
+			Timeout:        t.Timeout,
+			Retry:          t.Retry,
+			CircuitBreaker: t.CircuitBreaker,
+		}
+	}
+
+	for name, t := range buildingBlocks.Routes {
+		name = strings.TrimPrefix(name, "/") // Ignore slash prefix.
+		r.routes[name] = PolicyNames{
+			Timeout:        t.Timeout,
+			Retry:          t.Retry,
+			CircuitBreaker: t.CircuitBreaker,
+		}
+	}
+
+	return nil
+}
+
+// RoutePolicy returns the policy for a route.
+func (r *Resiliency) RoutePolicy(ctx context.Context, name string) Runner {
+	var t time.Duration
+	var rc *retry.Config
+	var cb *breaker.CircuitBreaker
+	name = strings.TrimPrefix(name, "/") // Ignore slash prefix.
+	operationName := fmt.Sprintf("route[%s]", name)
+	if r == nil {
+		return Policy(ctx, operationName, t, rc, cb)
+	}
+	policyNames, ok := r.routes[name]
+	if ok {
+		if policyNames.Timeout != "" {
+			t = r.timeouts[policyNames.Timeout]
+		}
+		if policyNames.Retry != "" {
+			rc = r.retries[policyNames.Retry]
+		}
+		if policyNames.CircuitBreaker != "" {
+			cb = r.circuitBreakers[policyNames.CircuitBreaker]
+		}
+	}
+
+	return Policy(ctx, operationName, t, rc, cb)
+}
+
+// RoutePolicy returns the policy for a service endpoint.
+func (r *Resiliency) EndpointPolicy(ctx context.Context, service string, endpoint string) Runner {
+	var t time.Duration
+	var rc *retry.Config
+	var cb *breaker.CircuitBreaker
+	operationName := fmt.Sprintf("endpoint[%s, %s]", service, endpoint)
+	if r == nil {
+		return Policy(ctx, operationName, t, rc, cb)
+	}
+	policyNames, ok := r.services[service]
+	if ok {
+		if policyNames.Timeout != "" {
+			t = r.timeouts[policyNames.Timeout]
+		}
+		if policyNames.Retry != "" {
+			rc = r.retries[policyNames.Retry]
+		}
+		if policyNames.CircuitBreaker != "" {
+			template := r.circuitBreakers[policyNames.CircuitBreaker]
+			ep, ok := r.serviceCBs[service]
+			if ok {
+				cb = ep.Get(endpoint, template)
+			}
+		}
+	}
+
+	return Policy(ctx, operationName, t, rc, cb)
+}
+
+// RemoveEndpoint removes an endpoint from a service.
+// This would be used when a service is taken out of service.
+func (r *Resiliency) RemoveEndpoint(name string, endpoint string) {
+	ep, ok := r.serviceCBs[name]
+	if ok {
+		ep.Remove(endpoint)
+	}
+}
+
+// ActorPolicy returns the policy for an actor instance.
+func (r *Resiliency) ActorPolicy(ctx context.Context, actorType string, id string) Runner {
+	var t time.Duration
+	var rc *retry.Config
+	var cb *breaker.CircuitBreaker
+	operationName := fmt.Sprintf("actor[%s, %s]", actorType, id)
+	if r == nil {
+		return Policy(ctx, operationName, t, rc, cb)
 	}
+	policyNames, ok := r.actors[actorType]
+	if ok {
```

Comment:
Does this not trip a linter? I feel like there's a nesting limit around 5.

Reply:
It probably does. I asked @artursouza to add `feature/*` to the github workflows to run this and the e2e tests. Those probably won't work until we merge master into this branch.

Reply:
This did not come up when running `make lint` but I fixed all the other issues that did :)